### PR TITLE
[Navigation Material] Update Docs and Sample to use new API for adding custom Navigators

### LINKED ISSUE #744: [Navigation] Integrate new rememberNavController APIs for custom Navigators
**Is your feature request related to a problem? Please describe.**
Navigation is adding an [API for adding custom Navigators](https://issuetracker.google.com/193555369) by extending the `rememberNavController()` API in the upcoming Navigation 2.4.0-alpha10 release.

**Describe the solution you'd like**
It would be nice if `rememberAnimatedNavController()` had those same APIs and the Navigation Material had its samples, documentation, and tests updated to use it / the new `rememberNavController()`.

**Describe alternatives you've considered**
This is just a convenience API, so you can do it manually, but keeping the API surfaces in sync between Navigation itself and Navigation Animation is super useful.

### PR DESCRIPTION
**Pull request by jossiwolf**
To be merged once Nav 2.4.0 Alpha 10 is released :)

Fixes #744

### CODE REVIEW

**Comment by ianhanniballake** [comment]
Navigation 2.4.0-alpha10 is out now so I think you can update the dependency now.

**Comment by jossiwolf** [comment]
Will do, thanks for the heads up! The snapshot was failing to resolve a dependency

**Comment by jossiwolf** [comment]
The one test that failed was a [Pager test](https://github.com/google/accompanist/pull/746/checks?check_run_id=3748866513), there shouldn't be any connection.

File: navigation-material/src/androidTest/java/com/google/accompanist/navigation.material/BottomSheetNavigatorTest.kt [hunk 1]
```diff
@@ -198,9 +198,9 @@ internal class BottomSheetNavigatorTest {
         val sheetState = ModalBottomSheetState(ModalBottomSheetValue.Hidden)
 
         composeTestRule.setContent {
-            navController = rememberNavController()
+            navigator = remember { BottomSheetNavigator(sheetState) }
+            navController = rememberNavController(navigator)
             if (savedState != null) navController.restoreState(savedState)
-            navigator = remember(navController) { BottomSheetNavigator(sheetState) }
             navController.navigatorProvider += navigator
```
Comment (ianhanniballake) [inline navigation-material/src/androidTest/java/com/google/accompanist/navigation.material/BottomSheetNavigatorTest.kt:204]: You don't need this line anymore

**Review by ianhanniballake — CHANGES_REQUESTED** [review]

File: navigation-material/src/androidTest/java/com/google/accompanist/navigation.material/BottomSheetNavigatorTest.kt (hunk 1 above)
Comment (jossiwolf) [reply navigation-material/src/androidTest/java/com/google/accompanist/navigation.material/BottomSheetNavigatorTest.kt:204]: Thanks 🙏

**Review by chrisbanes — APPROVED** [review]

**Pull request merged**